pkg/autodetect/main.go

```diff
@@ -147,6 +151,44 @@ func (b *Background) detectClusterRoles() {
 	}
 }
 
+func (b *Background) cleanDeployments() {
+	log.Debug("detecting orphaned deployments.")
+	deployments := &appsv1.DeploymentList{}
+	deployOpts := &client.ListOptions{}
+	jaegerOpts := &client.ListOptions{}
+	instances := &v1.JaegerList{}
+
+	// Only select fields that have the label 'sidecar.jaegertracing.io/inject'
+	if err := deployOpts.SetLabelSelector(inject.Annotation); err != nil {
```

Comment:
Is this actually working? I thought we had an annotation, not a label.

Reply:
Yes it works, I introduced a change to label the injected depolyments on the original PR when I used finalizers. 

https://github.com/jaegertracing/jaeger-operator/blob/master/pkg/inject/sidecar.go#L47

Reply:
Alright, sounds good. I think we kinda talked about this before, but let me cc @objectiser and @pavolloffay , as they might not be aware of this change.

@pavolloffay, @objectiser, TL;DR: we used to place the inject annotation under `annotations`, but we are now placing it under `labels`. The good thing is that we can now query deployments by the Jaeger Operator they are using (or simply by the existence of the label, like it's the case here). The downside is that we now have both an annotation (added by users) and a label (set by the operator).